### PR TITLE
fontforge 2017.07.31, 2015.04.31

### PR DESCRIPTION
**Pull request by peterdavidhamilton**
Add older versions of FontForge and zap preferences.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not [already refused].
- [ ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Thank you, but I’ll be rejecting as this has several issues (I’ll point them out in the file itself) and we don’t want support for older OSs that far back.

File: Casks/fontforge.rb [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
   appcast 'https://github.com/fontforge/fontforge/releases.atom',
           configuration: version.major_minor_patch.dots_to_hyphens
   name 'FontForge'
   homepage 'https://fontforge.github.io/en-US/'
 
-  depends_on macos: '>= :yosemite'
-
   app 'FontForge.app'
+
+  zap trash: [
```
Comment (vitorgalvao) [inline Casks/fontforge.rb]: Should be a string, not an array, because it’s a single entry. This one is still valid to be submitted in another PR.

File: Casks/fontforge.rb [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+  # https://github.com/fontforge/fontforge/releases/download/20170731/FontForge-20170730-Mac.dmg
+  elsif MacOS.version >= :yosemite
+    depends_on macos: '>= :yosemite'
+    version '2017.07.31'
+    sha256 '980c66e00f0f5c7a722eadd8b78a0ef2b20a7ccd952149cbfdfeaae9e66e4d4e'
+    url "https://github.com/fontforge/fontforge/releases/download/#{version.no_dots}/FontForge-#{version.no_dots.to_i - 1}-Mac.dmg"
+
+  # https://github.com/fontforge/fontforge/releases/download/20150430/FontForge-2015-04-30-Mac.app.dmg
+  else
+    depends_on macos: '>= :mavericks'
```
Comment (vitorgalvao) [inline Casks/fontforge.rb]: We don’t add these in OS version conditionals, as they makes no sense in this context.

File: Casks/fontforge.rb [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
-  version '2020.03.14.67687b0'
-  sha256 '394e4b1d216abd3d716dba94d9b6c7eacfdd9b3fa0c48d29fcaa1e27695593ce'
+  # https://github.com/fontforge/fontforge/releases/download/20200314/FontForge-2020-03-14-67687b0.app.dmg
+  if MacOS.version >= :sierra
+    depends_on macos: '>= :sierra'
+    version '2020.03.14.67687b0'
+    sha256 '394e4b1d216abd3d716dba94d9b6c7eacfdd9b3fa0c48d29fcaa1e27695593ce'
+    url "https://github.com/fontforge/fontforge/releases/download/#{version.major_minor_patch.no_dots}/FontForge-#{version.dots_to_hyphens}.app.dmg"
+
+  # https://github.com/fontforge/fontforge/releases/download/20170731/FontForge-20170730-Mac.dmg
```
Comment (vitorgalvao) [inline Casks/fontforge.rb]: We don’t want these hanging comments. If they had to be updated with the `url`, they would defeat the point of the interpolation.

File: Casks/fontforge.rb [hunk 4]
```diff
@@ -1,15 +1,35 @@
 cask 'fontforge' do
-  version '2020.03.14.67687b0'
-  sha256 '394e4b1d216abd3d716dba94d9b6c7eacfdd9b3fa0c48d29fcaa1e27695593ce'
+  # https://github.com/fontforge/fontforge/releases/download/20200314/FontForge-2020-03-14-67687b0.app.dmg
+  if MacOS.version >= :sierra
```
Comment (vitorgalvao) [inline Casks/fontforge.rb]: OS conditionals of this type should always be `<=`.

File: Casks/fontforge.rb [hunk 5]
```diff
@@ -1,15 +1,35 @@
...
+    depends_on macos: '>= :sierra'
```
Comment (vitorgalvao) [inline Casks/fontforge.rb]: This is essentially a repeat of the `if`.

File: Casks/fontforge.rb [hunk 6]
```diff
@@ -1,15 +1,35 @@
...
+    version '2020.03.14.67687b0'
+    sha256 '394e4b1d216abd3d716dba94d9b6c7eacfdd9b3fa0c48d29fcaa1e27695593ce'
+    url "https://github.com/fontforge/fontforge/releases/download/#{version.major_minor_patch.no_dots}/FontForge-#{version.dots_to_hyphens}.app.dmg"
+
+  # https://github.com/fontforge/fontforge/releases/download/20170731/FontForge-20170730-Mac.dmg
+  elsif MacOS.version >= :yosemite
+    depends_on macos: '>= :yosemite'
+    version '2017.07.31'
+    sha256 '980c66e00f0f5c7a722eadd8b78a0ef2b20a7ccd952149cbfdfeaae9e66e4d4e'
+    url "https://github.com/fontforge/fontforge/releases/download/#{version.no_dots}/FontForge-#{version.no_dots.to_i - 1}-Mac.dmg"
```
Comment (vitorgalvao) [inline Casks/fontforge.rb]: Clever with the `- 1`, but we don’t do that. These versions are different likely due to a problem that will be fixed.

Comment (peterdavidhamilton) [reply Casks/fontforge.rb]: Thanks for taking the time to explain the issues Vitor. I will do a new PR just for the zap.
